### PR TITLE
[PM-7288] Include changes to actions in scan workflow

### PR DESCRIPTION
**Pull request by bwdil**
- Upload CheckMarx results in cx_results.sarif to GitHub
  - Arguments to support code quality scan with SonarCloud
  - Add SonarCloud to Gradle Build process to support scan

## 🎟️ Tracking

[PM-7288 ](https://bitwarden.atlassian.net/browse/PM-7288) - Deploy scanning workflow organization-wide 

## 📔 Objective

As part of our efforts to support secure code quality initiatives and find vulnerabilities through the use CheckMarx and SonarCloud, we need to implement these changes to our scan workflow. The workflows may change over time as the AppSec team further refines the process and workflow.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

None

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -7,4 +7,13 @@ plugins {
...
+    id("org.sonarqube") version "5.0.0.4638"
+}
+
+sonar {
+    properties {
+        property("sonar.projectKey", "bitwarden_authenticator-android")
+        property("sonar.organization", "bitwarden")
+        property("sonar.host.url", "https://sonarcloud.io")
+    }
 }
```
Comment (withinfocus) [inline build.gradle.kts:19]: ⛏️ Add a newline while you're here.

File: .github/workflows/scan.yml [hunk 2]
```diff
@@ -68,8 +71,8 @@ jobs:
           SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         with:
-          # TODO: see https://docs.sonarsource.com/sonarcloud/advanced-setup/analysis-parameters/
-          # for parameters to configure
           args: >
             -Dsonar.organization=${{ github.repository_owner }}
             -Dsonar.projectKey=${{ github.repository_owner }}_${{ github.event.repository.name }}
+            -Dsonar.sources=app/src/main,app/src/debug
+            -Dsonar.tests=app/src/test,app/src/androidTest
```
Comment (withinfocus) [inline .github/workflows/scan.yml:78]: 🎨 Like in other repos we should perform all the configuration in one place, either here or in the Gradle config; seems like the latter is what's needed.

**Review by withinfocus — CHANGES_REQUESTED** [review]
We also need the mobile team to confirm the paths included here.

File: build.gradle.kts (hunk 1 above)
Comment (bwdil) [reply build.gradle.kts]: Done

File: .github/workflows/scan.yml (hunk 2 above)
Comment (bwdil) [reply .github/workflows/scan.yml]: Done and noted

Comment (bwdil) [reply .github/workflows/scan.yml]: I moved the source and test paths over to the gradle build script, but I didn't see this in the sonar documentation, so hopefully they belong there. I guess it's the same as including them as arguments.

**Review by SaintPatrck — CHANGES_REQUESTED** [review]

File: build.gradle.kts [hunk 3]
```diff
@@ -7,4 +7,16 @@ plugins {
     alias(libs.plugins.kotlinx.kover) apply false
     alias(libs.plugins.ksp) apply false
     alias(libs.plugins.google.services) apply false
-}
\ No newline at end of file
+    id("org.sonarqube") version "5.0.0.4638"
```
[the remark this answers is not in the document]
Comment (bwdil) [reply build.gradle.kts]: Done and thank you for the feedback @SaintPatrck

File: build.gradle.kts [hunk 4]
```diff
@@ -7,4 +7,16 @@ plugins {
...
+
+sonar {
+    properties {
+        property("sonar.projectKey", "bitwarden_authenticator-android")
+        property("sonar.organization", "bitwarden")
+        property("sonar.host.url", "https://sonarcloud.io")
+        property("sonar.sources", "app/src/main/,app/src/debug/")
+        property("sonar.tests", "app/src/test/,app/src/androidTest/")
+    }
+}
```
[the remark this answers is not in the document]
Comment (bwdil) [reply build.gradle.kts]: Done.

File: build.gradle.kts [hunk 5]
```diff
@@ -7,4 +7,16 @@ plugins {
...
+    id("org.sonarqube") version "5.0.0.4638"
+}
+
+sonar {
+    properties {
+        property("sonar.projectKey", "bitwarden_authenticator-android")
+        property("sonar.organization", "bitwarden")
+        property("sonar.host.url", "https://sonarcloud.io")
+        property("sonar.sources", "app/src/main/,app/src/debug/")
+        property("sonar.tests", "app/src/test/,app/src/androidTest/")
```
[the remark this answers is not in the document]
Comment (bwdil) [reply build.gradle.kts]: Done.

File: build.gradle.kts [hunk 6]
```diff
@@ -7,4 +7,16 @@ plugins {
     alias(libs.plugins.kotlinx.kover) apply false
     alias(libs.plugins.ksp) apply false
     alias(libs.plugins.google.services) apply false
-}
\ No newline at end of file
+    id("org.sonarqube") version "5.0.0.4638"
+}
+
+sonar {
```
[the remark this answers is not in the document]
Comment (bwdil) [reply build.gradle.kts]: Done.

**Review by SaintPatrck — DISMISSED** [review]

**Review by withinfocus — DISMISSED** [review]

**Review by SaintPatrck — DISMISSED** [review]

**Comment by bwdil** [comment]
@bitwarden/dept-devops can I get a review and approval?

**Review by SaintPatrck — APPROVED** [review]

**Review by withinfocus — APPROVED** [review]
Duplicative configs but it's what's needed.

**Review by Eeebru — APPROVED** [review]

**Pull request merged**